### PR TITLE
[3.x] Update Node 20.12.2, Npm 10.5.0 and Typescript 5.4.5 (Dev bundle 20.12.2.0)

### PR DESCRIPTION
**Pull request by nachocodoner**
OSS-386
OSS-385

-----

Summary of this PR [here](https://github.com/meteor/meteor/pull/13090#issuecomment-2113923381)

### Fix on Windows tests

After the 20.12.2 node.js updates, calling `spawn` with a .cmd or .bat file requires the `shell: true` option. This PR adjusts our usage accordingly.

#### Update

Tests are fixed on Windows as seen on this picture:
image

However, it doesn't finish properly [reaching this code](https://github.com/meteor/meteor/blob/typescript-5.4.4/scripts/windows/appveyor/test.ps1#L22-L26), and it times out at the end.

And locally, I can't seem to reproduce the same, locally it finishes properly.

image

Trying to debug the code around spawn of child processes, since maybe one is never stop after last adjusts. But hard to verify it since locally behaves differently.

### CODE REVIEW

**Comment by Grubba27** [comment]
you need to build a new dev bundle btw

File: meteor [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-BUNDLE_VERSION=20.11.1.2
+BUNDLE_VERSION=20.11.1.3
```
Comment (Grubba27) [inline meteor:3]: it should be: `20.12.2.0`

File: scripts/dev-bundle-tool-package.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ var packageJson = {
     npm: "10.2.4",
```
Comment (Grubba27) [inline scripts/dev-bundle-tool-package.js:13]: missing updating here to `10.5.0`

**Comment by StorytellerCZ** [comment]
Node 20.13.0 is out:
https://github.com/nodejs/node/releases/tag/v20.13.0

**Comment by StorytellerCZ** [comment]
The latest release of Node has some reverse on Windows stuff:
https://github.com/nodejs/node/releases/tag/v20.13.1

**Comment by nachocodoner** [comment]
Thanks! Once CI responds sucessfully after recent Windows environment failures, we can proceed to 20.13.1. There are still unresolved issues with using the `shell: true` option for spawn scripts in a Windows environment, this has not been reverted, then self-test Meteor must continue to pass successfully.

The current problem is that CI environments don't consistently give the same success as local environments. We consider migrating to GitHub Actions using a custom runner to have greater control over the environment.

**Comment by nachocodoner** [comment]
Tests pass 🟢!

To summarize what happened in this PR:

- In the Node `20.12.2` upgrade there is a breaking change on Windows environment requiring to adapt meteor tool on Windows and spawn of processes with `shell: true`. Check how another library had [to adapt the breaking change similarly](https://github.com/node-red/node-red/pull/4652/files#diff-f45887bf975d28302fb92240b7e62b250a93e5746557ac4bf70c2407cb51b954R584).
- One of our CI checks started to fail, and even after fixing the breaking change, the next error araised was a timedout consistently. Some of the work then we do next is to ensure our CI tests signal green, the same way it happened locally in my Windows machine. The phases on this work:
	- AppVeyor check. [It triggered timeout](https://ci.appveyor.com/project/meteor/meteor/builds/49634541/messages), the process was reaching the limits. No plan to upgrade since we transit generally to GitHubActions.
	- GitHub action migration using their runners. [One test didn't pass green](https://github.com/meteor/meteor/actions/runs/8879483823/job/24377301961?pr=13090), a specific test when testing the client part in headless mode. Likely a limitation on Github runners so we prefered to opt for other solution.
	- GitHub action migration using custom runner on a AWS machine. Having more control on the machine set up and replicate what we got locally. After several configs, [it passed ok at last](https://github.com/meteor/meteor/actions/runs/9098831119/job/25009957177?pr=13090).

**Review by StorytellerCZ — APPROVED** [review]

**Review by denihs — APPROVED** [review]

File: tools/cli/main.js [hunk 3]
```diff
@@ -552,7 +552,8 @@ var springboard = async function (rel, options) {
       var execPath = files.convertToOSPath(executable);
       var child = require("child_process").spawn(execPath, newArgv, {
         env: process.env,
-        stdio: 'inherit'
+        stdio: 'inherit',
+        shell: true,
```
Comment (henriquealbert) [inline tools/cli/main.js:556]: dont you need to check the win32 platform here as well like you did [here](https://github.com/meteor/meteor/pull/13090/files#diff-2f3c40a81aa9a3dfb04ee79c789ff27df4ca927c3247279897f8ed9066a7c628R73)?

Comment (nachocodoner) [reply tools/cli/main.js:556]: image

The check is outer already.

**Review by denihs — APPROVED** [review]

**Pull request merged**